scripts/azureml-assets/azureml/assets/release/asset_publish.py

```diff
@@ -142,27 +250,57 @@ def _str2bool(v: str) -> bool:
             print(cmd)
 
             # Run command
-            if debug_mode:
-                # Capture and redact output
-                results = run(cmd, stdout=PIPE, stderr=STDOUT, encoding=sys.stdout.encoding, errors="ignore")
-                redacted_output = re.sub(r"Bearer.*", "", results.stdout)
-                logger.print(redacted_output)
-            else:
-                results = run(cmd)
-
-            # Check command result
-            if results.returncode != 0:
-                logger.log_warning(f"Error creating {asset}")
-                failure_list.append(asset)
+            results = run_command(cmd, failure_list, debug_mode)
+
+        elif asset.type == assets.AssetType.MODEL:
```

Comment:
Since I have to run the run_command within the context_manager for the model, I will create function for preparing and executing the command and run it within the if else.